packages/worker/src/worker.ts

```diff
@@ -1746,6 +1742,14 @@ async function extractActivityInfo(
   const { taskToken } = task as NonNullableObject<coresdk.activity_task.IActivityTask>;
   const start = task.start as NonNullableObject<coresdk.activity_task.IStart>;
   const activityId = start.activityId;
+  let heartbeatDetails = undefined;
+  try {
+    heartbeatDetails = await decodeFromPayloadsAtIndex(dataConverter, 0, start.heartbeatDetails);
+  } catch (e) {
+    throw ApplicationFailure.fromError(e, {
+      message: `Failed to parse heartbeat details for activity ${activityId}: ${errorMessage(e)}`,
+    });
```

Comment:
Is this non-retriable?

Reply:
the default is retryable.

Reply:
Sorry, I meant whether we should make it non-retryable and fail the workflow

Reply:
Payload conversion error could be due to transient issues.